src/stacksResolver.ts

```diff
@@ -0,0 +1,21 @@
+import { IResolver } from 'jolocom-lib/js/didMethods/types'
+import { Identity } from 'jolocom-lib/js/identity/identity'
+import { getResolver } from '@jolocom/stacks-did-resolver'
+import { DidDocument } from 'jolocom-lib/js/identity/didDocument/didDocument'
+import { ErrorCodes } from 'jolocom-lib/js/errors'
+import { PREFIX } from './constant'
+
+export class StacksResolver implements IResolver {
+  prefix = PREFIX
+
+  async resolve(did: string): Promise<Identity> {
+    const jsonDidDoc = await getResolver()(did).catch(() => {
```

Comment:
Calling `getResolver` constructs a new resolver each time. 
This should rather be done during actual construction (i.e. in the factory)
Also please add a `StacksNetwork` parameter to the factory, to be passed down to `getResolver`

Reply:
Please check now if that's ok

Reply:
The way you refactored it still causes `getResolver` to be called each time